### PR TITLE
Publish curated recipes

### PR DESCRIPTION
**Pull request by annebyrne**
Change publishing so that curated recipes are immediately pushed to CAPI rather than waiting for 3 step verification process. To allow greater amount of data for testing.

### CODE REVIEW

File: ui/app/com/gu/recipeasy/AppComponents.scala [hunk 1]
```diff
@@ -61,14 +61,13 @@ class AppComponents(context: Context)
...
   val healthcheckController = new Healthcheck
-  val applicationController = new Application(wsClient, configuration, db, messagesApi)
 
   val publisherController = {
     val publisherConfig = PublisherConfig(configuration, region, identity.stage)
     val contentApiClient = new ContentApi(contentApiClient = new GuardianContentClient(publisherConfig.contentAtomConfig.capiKey))
     new Publisher(wsClient, configuration, publisherConfig, db, teleporter, contentApiClient)
   }
-
+  val applicationController = new Application(wsClient, configuration, db, messagesApi, publisherController)
```
Comment (LATaylor-guardian) [inline ui/app/com/gu/recipeasy/AppComponents.scala]: From a design point of view, it doesn't feel right for one controller to have a dependency on another controller, but I can see why you've done it that way. I'd recommend creating a new object called something like `RecipePublisher` and moving the logic to publish from the `PublishController` inside of it. That way Both ApplicationController and PublisherController can share the use of it and you remove the need for one controller to depend on the other. Controllers are more you entry points for your routes from your API.

File: ui/app/com/gu/recipeasy/AppComponents.scala [hunk 2]
```diff
@@ -61,13 +58,14 @@ class AppComponents(context: Context)
...
-  val publisherController = {
-    val publisherConfig = PublisherConfig(configuration, region, identity.stage)
-    val contentApiClient = new ContentApi(contentApiClient = new GuardianContentClient(publisherConfig.contentAtomConfig.capiKey))
-    new Publisher(wsClient, configuration, publisherConfig, db, teleporter, contentApiClient)
-  }
+
+  val publisherConfig = PublisherConfig(configuration, region, identity.stage)
+  val contentApiClient = new ContentApi(contentApiClient = new GuardianContentClient(publisherConfig.contentAtomConfig.capiKey))
+
+  val publisherController = new Publisher(wsClient, configuration, publisherConfig, db, teleporter, contentApiClient)
```
Comment (mchv) [inline ui/app/com/gu/recipeasy/AppComponents.scala]: why `publisherConfig` and `contentApiClient ` need to be exposed?

File: ui/app/com/gu/recipeasy/controllers/Publisher.scala [hunk 3]
```diff
@@ -1,51 +1,56 @@
...
             }
-
+            Future.successful(Ok(s"Publishing content atom"))
+          }
           case None => Future.successful(NotFound)
         }
+      }
       case None => Future.successful(NotFound)
     }
   }
```
Comment (mchv) [inline ui/app/com/gu/recipeasy/controllers/Publisher.scala]:
```scala
  private def send(internalComposerCode: String, recipe: Recipe, curatedRecipe: CuratedRecipe, curatedRecipeImages: Seq[Image])(config: PublisherConfig) = {

    val atomEvents: (AuxiliaryAtomEvent, ContentAtomEvent) = {
      val contentAtom = CuratedRecipe.toAtom(recipe, curatedRecipe, curatedRecipeImages)
      val auxiliaryAtomEvent = AuxiliaryAtomEvent(internalComposerCode, eventType = AuxiliaryAtomEventType.Add, Seq(AuxiliaryAtom(contentAtom.id, "recipe")))
      val contentAtomEvent = ContentAtomEvent(contentAtom, EventType.Update, eventCreationTime = OffsetDateTime.now.toInstant.toEpochMilli)
      (auxiliaryAtomEvent, contentAtomEvent)
    }

    AtomPublisher.send(atomEvents)(config)
  }
``` 

could be deleted now it has moved to `RecipePublisher`

File: ui/app/com/gu/recipeasy/AppComponents.scala (hunk 2 above)
Comment (LATaylor-guardian) [inline ui/app/com/gu/recipeasy/AppComponents.scala]: Sorry my bad, I wrote code in github earlier that wasn't clear on this.

**Comment by mchv** [comment]
👍

**Comment by shtukas** [comment]
I am closing this because of git branch confusion. Anne is going to resubmit this work on branch `publish-curated-recipes` and I will :+1: reporting Mariot's :+1:

**Pull request closed without merging**